ui/src/components/payload_editor/PayloadEditor.js

```diff
@@ -0,0 +1,71 @@
+import React, {Component} from "react";
+import AceEditor from "react-ace";
+import Row from "react-bootstrap/Row";
+import Col from "react-bootstrap/Col";
+
+import "ace-builds/src-noconflict/mode-json";
+import "ace-builds/src-noconflict/theme-xcode";
+
+export class PayloadEditor extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      didChange: false,
+      newCode: undefined,
+      editorCode: ""
+    };
+
+    this.didIntroduceChange = this.didIntroduceChange.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+  didIntroduceChange() {
```

Comment:
Again, naming confuses me. This feels like a function that returns a Boolean value, while `this.state.newCode` probably holds something else?